iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber_uds.inl

```diff
@@ -0,0 +1,146 @@
+    SampleType& sample = *samplePtr;
+
+    std::string tempBuffer;
+
+    // Receive the first (up to) 4095 Bytes
+    m_uds.receive().and_then([&](auto& msg) { tempBuffer.append(msg); }).or_else([](auto&) {
+        std::cerr << "Error occurred while receiving UNIX domain socket!" << std::endl;
+    });
+
+    if (tempBuffer.size() == 0)
+    {
+        return 0;
+    }
+
+    // Deserialize the counter
+    sample.counter =
+        (static_cast<uint32_t>(tempBuffer[3]) & 0xFF) << 24 | (static_cast<uint32_t>(tempBuffer[2]) & 0xFF) << 16
+        | (static_cast<uint32_t>(tempBuffer[1]) & 0xFF) << 8 | (static_cast<uint32_t>(tempBuffer[0]) & 0xFF);
+
+    // Deserialize the timestamp
+    uint64_t sendTimestamp =
+        (static_cast<uint64_t>(tempBuffer[11]) & 0xFF) << 56 | (static_cast<uint64_t>(tempBuffer[10]) & 0xFF) << 48
+        | (static_cast<uint64_t>(tempBuffer[9]) & 0xFF) << 40 | (static_cast<uint64_t>(tempBuffer[8]) & 0xFF) << 32
+        | (static_cast<uint64_t>(tempBuffer[7]) & 0xFF) << 24 | (static_cast<uint64_t>(tempBuffer[6]) & 0xFF) << 16
+        | (static_cast<uint64_t>(tempBuffer[5]) & 0xFF) << 8 | (static_cast<uint64_t>(tempBuffer[4]) & 0xFF);
+    int64_t castedSendTimestamp = static_cast<int64_t>(sendTimestamp);
+    sample.sendTimestamp = std::chrono::time_point<std::chrono::steady_clock>(
+        std::chrono::duration<int64_t, std::nano>(castedSendTimestamp));
+
+    // Deserialize subPackets
+    uint32_t subPackets =
+        (static_cast<uint32_t>(tempBuffer[15]) & 0xFF) << 24 | (static_cast<uint32_t>(tempBuffer[14]) & 0xFF) << 16
+        | (static_cast<uint32_t>(tempBuffer[13]) & 0xFF) << 8 | (static_cast<uint32_t>(tempBuffer[12]) & 0xFF);
+    sample.subPackets = subPackets;
+
+    // If more than 4095 Bytes were send in consecutive messages, receive them now
+    if (subPackets > 1)
+    {
+        for (uint32_t i = 0U; i < subPackets - 1; ++i)
+        {
+            m_uds.receive().and_then([&](auto& msg) { tempBuffer.append(msg); }).or_else([](auto&) {
+                std::cerr << "Error occurred while receiving UNIX domain socket!" << std::endl;
+            });
+        }
+    }
+
+    // Complete fragmented message was received, now we call the user-defined callable
+    callable(samplePtr);
+    return numberOfSamples;
+}
+
+template <typename T>
+inline void EventSubscriberUds<T>::SetReceiveHandler(EventReceiveHandler handler)
+{
+    std::lock_guard<iox::posix::mutex> guard(m_mutex);
+    if (HasReceiveHandler())
+    {
+        std::cout << "Re-attaching a receiver handler is not supported with UNIX domain sockets!" << std::endl;
+        return;
+    }
+    m_receiveHandler.emplace(handler);
+    std::thread([&]() {
+        while (true)
+        {
+            // We call the user callback in an endless loop and wait till having received a complete message
+            m_receiveHandler.and_then([](iox::cxx::function<void()>& userCallable) { userCallable(); });
```

Comment:
`userCallable` is nullable therefore you have to verify that it is not null before calling it.